packages/babel-traverse/src/scope/binding.ts

```diff
@@ -44,6 +44,13 @@ export default class Binding {
     this.path = path;
     this.kind = kind;
 
+    for (let node of path.getAncestry()) {
+      if (node.isLoop()) {
+        debugger;
+        this.constant = false;
+      }
+    }
```

Comment:
1. When `node.isFunction()` we can break the loop, since variables don't cross function boundaries:
   ```js
   for (;;) () => { var x };
   ```
   is constant.
2. We only need to do this if we have a `var`, not for `const` or `let`
3. We don't need to collect the whole ancestry, since we'll only use part of it (up to a function). We can do something like
   ```js
   let { parentPath } = path;
   while (parentPath && !parentPath.isFunction()) {
     // ...
     ({ parentPath } = parentPath);
   }
   ```